### DISCUSSION TITLE
next-with-apollo: Hook running on server

### DISCUSSION THREAD

**Discussion by amorino** [vercel/next.js#21202] [Help]
I'm using the example `next-with-apollo` to setup a project with Apollo, the problem that when this line of code is called on the server:

https://github.com/vercel/next.js/blob/90ad2cbc44f5e807d7507c1eecc73ceb5445b6dc/examples/with-apollo/pages/_app.js#L5

I put up an Auth link using the following code and when I used it on `getServerSideProps` I don't have any problems, the `ctx` exists.

```tsx
const httpLink = new HttpLink({
    uri: config.API_URL,
    credentials: 'same-origin',
  });

  const authLink = setContext((_, { headers }) => {
    // Get the authentication token from cookies
    const { token } = parseCookies(ctx);

    return {
      headers: {
        ...headers,
        authorization: token ? `Bearer ${token}` : '',
      },
    };
  });
```

However I don't understand how a line of code inside a component and a hook is getting called on the server.

**Comment by ka8725** [comment]
There are several ways how to "inject" apollo client instance into components. Seems you go with `_app.js` approach. Please, follow the docs [there](https://github.com/lfades/next-with-apollo#using-_app).
